Citra can't be built without SDL support.

Git Commit : `935bcdbd204bfc1d601fd5d12851ded188b69266`
OS : Gentoo Linux

I tried to compile Citra without SDL support on Gentoo, and I encountered the following error.

```
/var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/dedicated_room/citra-room.cpp:12:10: fatal error: 'SDL.h' file not found
#include <SDL.h>
```

libsdl and libsdl2 are installed on my system, though.

I enabled sdl2 and compiled citra with gcc-7.3.0. I got the following errors

```
/var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/core/file_sys/archive_backend.cpp: In constructor ‘FileSys::Path::Path(FileSys::LowPathType, const std::vector<unsigned char>&)’:
/var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/core/file_sys/archive_backend.cpp:24:32: error: invalid conversion from ‘const void*’ to ‘void*’ [-fpermissive]
         std::memcpy(string.data(), data.data(), string.size());
                     ~~~~~~~~~~~^~
In file included from /usr/include/features.h:410:0,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/6.4.0/include/g++-v6/x86_64-pc-linux-gnu/bits/os_defines.h:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/6.4.0/include/g++-v6/x86_64-pc-linux-gnu/bits/c++config.h:507,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/6.4.0/include/g++-v6/cstddef:49,
                 from /var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/core/file_sys/archive_backend.cpp:5:
/usr/include/bits/string3.h:50:1: note:   initializing argument 1 of ‘void* memcpy(void*, const void*, size_t)’
 __NTH (memcpy (void *__restrict __dest, const void *__restrict __src,
 ^
/var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/core/file_sys/archive_backend.cpp:30:32: error: invalid conversion from ‘const void*’ to ‘void*’ [-fpermissive]
         std::memcpy(u16str.data(), data.data(), u16str.size() * sizeof(char16_t));
                     ~~~~~~~~~~~^~
In file included from /usr/include/features.h:410:0,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/6.4.0/include/g++-v6/x86_64-pc-linux-gnu/bits/os_defines.h:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/6.4.0/include/g++-v6/x86_64-pc-linux-gnu/bits/c++config.h:507,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/6.4.0/include/g++-v6/cstddef:49,
                 from /var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/core/file_sys/archive_backend.cpp:5:
/usr/include/bits/string3.h:50:1: note:   initializing argument 1 of ‘void* memcpy(void*, const void*, size_t)’
 __NTH (memcpy (void *__restrict __dest, const void *__restrict __src,
 ^
make[2]: *** [src/core/CMakeFiles/core.dir/build.make:399: src/core/CMakeFiles/core.dir/file_sys/archive_backend.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/var/tmp/portage/games-emulation/citra-9999/work/citra-9999_build'
make[1]: *** [CMakeFiles/Makefile2:678: src/core/CMakeFiles/core.dir/all] Error 2
make[1]: Leaving directory '/var/tmp/portage/games-emulation/citra-9999/work/citra-9999_build'
make: *** [Makefile:141: all] Error 2
 * ERROR: games-emulation/citra-9999::jm-overlay failed (compile phase):
 *   emake failed
```

I chose gcc-7.3.0 with `gcc-config` and tried to compile citra again. Now, I see new errors.

```
In file included from /usr/include/boost/icl/type_traits/identity_element.hpp:11:0,
                 from /usr/include/boost/icl/type_traits/unit_element.hpp:12,
                 from /usr/include/boost/icl/concept/interval.hpp:17,
                 from /usr/include/boost/icl/concept/joinable.hpp:12,
                 from /usr/include/boost/icl/associative_interval_container.hpp:13,
                 from /usr/include/boost/icl/interval_base_set.hpp:24,
                 from /usr/include/boost/icl/interval_set.hpp:14,
                 from /usr/include/boost/icl/interval_map.hpp:13,
                 from /var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/./video_core/renderer_opengl/gl_rasterizer_cache.h:15,
                 from /var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/./video_core/renderer_opengl/gl_rasterizer.h:26,
                 from /var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/video_core/renderer_opengl/gl_shader_gen.cpp:16:
/usr/include/boost/icl/type_traits/type_to_string.hpp:56:12: error: partial specialization of ‘struct boost::icl::type_to_string<Unary<Type> >’ after instantiation of ‘struct boost::icl::type_to_string<std::__cxx11::basic_string<char> >’ [-fpermissive]
     struct type_to_string<Unary<Type> >
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/boost/icl/type_traits/type_to_string.hpp:72:12: error: partial specialization of ‘struct boost::icl::type_to_string<Binary<Type1, Type2> >’ after instantiation of ‘struct boost::icl::type_to_string<std::__cxx11::basic_string<char> >’ [-fpermissive]
     struct type_to_string<Binary<Type1, Type2> >
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/boost/icl/type_traits/identity_element.hpp:11:0,
                 from /usr/include/boost/icl/type_traits/unit_element.hpp:12,
                 from /usr/include/boost/icl/concept/interval.hpp:17,
                 from /usr/include/boost/icl/concept/joinable.hpp:12,
                 from /usr/include/boost/icl/associative_interval_container.hpp:13,
                 from /usr/include/boost/icl/interval_base_set.hpp:24,
                 from /usr/include/boost/icl/interval_set.hpp:14,
                 from /usr/include/boost/icl/interval_map.hpp:13,
                 from /var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/./video_core/renderer_opengl/gl_rasterizer_cache.h:15,
                 from /var/tmp/portage/games-emulation/citra-9999/work/citra-9999/src/video_core/renderer_opengl/gl_rasterizer_cache.cpp:28:
/usr/include/boost/icl/type_traits/type_to_string.hpp:56:12: error: partial specialization of ‘struct boost::icl::type_to_string<Unary<Type> >’ after instantiation of ‘struct boost::icl::type_to_string<std::__cxx11::basic_string<char> >’ [-fpermissive]
     struct type_to_string<Unary<Type> >
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/boost/icl/type_traits/type_to_string.hpp:72:12: error: partial specialization of ‘struct boost::icl::type_to_string<Binary<Type1, Type2> >’ after instantiation of ‘struct boost::icl::type_to_string<std::__cxx11::basic_string<char> >’ [-fpermissive]
     struct type_to_string<Binary<Type1, Type2> >
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/video_core/CMakeFiles/video_core.dir/build.make:231: src/video_core/CMakeFiles/video_core.dir/renderer_opengl/gl_rasterizer.cpp.o] Error 1
make[2]: *** [src/video_core/CMakeFiles/video_core.dir/build.make:279: src/video_core/CMakeFiles/video_core.dir/renderer_opengl/gl_shader_gen.cpp.o] Error 1
make[2]: *** [src/video_core/CMakeFiles/video_core.dir/build.make:255: src/video_core/CMakeFiles/video_core.dir/renderer_opengl/gl_rasterizer_cache.cpp.o] Error 1
make[2]: Leaving directory '/var/tmp/portage/games-emulation/citra-9999/work/citra-9999_build'
make[1]: *** [CMakeFiles/Makefile2:730: src/video_core/CMakeFiles/video_core.dir/all] Error 2
make[1]: Leaving directory '/var/tmp/portage/games-emulation/citra-9999/work/citra-9999_build'
make: *** [Makefile:141: all] Error 2
 * ERROR: games-emulation/citra-9999::jm-overlay failed (compile phase):
 *   emake failed
```

--- a/src/dedicated_room/CMakeLists.txt
+++ b/src/dedicated_room/CMakeLists.txt
@@ -12,7 +12,7 @@ target_link_libraries(citra-room PRIVATE glad)
 if (MSVC)
     target_link_libraries(citra-room PRIVATE getopt)
 endif()
-target_link_libraries(citra-room PRIVATE ${PLATFORM_LIBRARIES} SDL2 Threads::Threads)
+target_link_libraries(citra-room PRIVATE ${PLATFORM_LIBRARIES} Threads::Threads)
 
 if(UNIX AND NOT APPLE)
     install(TARGETS citra-room RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")

--- a/src/dedicated_room/citra-room.cpp
+++ b/src/dedicated_room/citra-room.cpp
@@ -8,8 +8,6 @@
 #include <regex>
 #include <string>
 #include <thread>
-#define SDL_MAIN_HANDLED
-#include <SDL.h>
 #include <glad/glad.h>
 
 #ifdef _MSC_VER
@@ -60,7 +58,7 @@ int main(int argc, char** argv) {
     char* endarg;
 
     // This is just to be able to link against core
-    gladLoadGLLoader(static_cast<GLADloadproc>(SDL_GL_GetProcAddress));
+    gladLoadGL();
 
     std::string room_name;
     std::string password;

